packages/at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -501,6 +504,9 @@ class NotificationServiceImpl
       stopAllSubscriptions();
       _logger.finer(
           'removing from _notificationServiceMap: ${_atClient.getCurrentAtSign()}');
+      _atClientManager.removeChangeListeners(
```

Comment:
Unable to comment on an if condition above, just for my understanding, why is the if checking for previous and present atSign to be same?

if (switchAtSignEvent.previousAtClient?.getCurrentAtSign() ==
        _atClient.getCurrentAtSign()) {

Reply:
In the above IF condition, _atClient.getCurrentAtSign returns the atSign with which the `NotificationServiceImpl` is created. Since we only wanted to remove the previous atSign (not the new atSign) added the above IF condition.

Reply:
Understood the code. Pls add dart docs if it makes sense.